engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -528,9 +577,12 @@ public BlockRegion union(EntityRef blockRef) {
     // ---------------------------------------------------------------------------------------------------------------//
 
     /**
-     * calculate the BlockRegion that is intersected between another region
+     * Compute the intersection of this region with the {@code other} region.
+     * <p>
+     * NOTE: If the regions don't intersect this region will become invalid!
```

Comment:
This bit is a bit nasty, as this is currently the only way how to get an invalid region I think...

Reply:
I don't think you can avoid everything.